drivers/SmartThings/philips-hue/src/handlers/lifecycle_handlers/button.lua

```diff
@@ -150,31 +147,14 @@ function ButtonLifecycleHandlers.init(driver, device)
       end
     end
   end
-  local svc_rids_for_device = driver.services_for_device_rid[hue_device_id] or {}
-  if button_info and button_info.num_buttons == nil and not svc_rids_for_device[button_info.id]
-  then
-    svc_rids_for_device[button_info.id] = HueDeviceTypes.BUTTON
-  end
-
-  if button_info and button_info.num_buttons then
-    for var = 1, (button_info.num_buttons or 1) do
-      local button_id_key = string.format("button%s_id", var)
-      local button_id = button_info[button_id_key]
-      svc_rids_for_device[button_id] = HueDeviceTypes.BUTTON
-    end
-  end
-
-  if button_info and not svc_rids_for_device[button_info.power_id] then
-    svc_rids_for_device[button_info.power_id] = HueDeviceTypes.DEVICE_POWER
-  end
-
-  driver.services_for_device_rid[hue_device_id] = svc_rids_for_device
-  log.debug(st_utils.stringify_table(driver.services_for_device_rid[hue_device_id], "svcs for device rid", true))
-  for rid, _ in pairs(driver.services_for_device_rid[hue_device_id]) do
-    log.debug(string.format("Button %s mapping to [%s]", device.label, rid))
-    driver.hue_identifier_to_device_record[rid] = device
+  if not button_info then
+    log.warn(string.format("Button %s parent bridge not ready, queuing refresh", device and device.label))
+    driver._devices_pending_refresh[device.id] = device
```

Comment:
Why is this situation only possible for Contact and motion sensors? Is it related to the multiservice situation with these devices?

Reply:
[This code](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/1dbda0f268167b54c1b1e625ae31315ffbca62fd/drivers/SmartThings/philips-hue/src/handlers/lifecycle_handlers/light.lua#L126-L136) could probably be refactored to be the same, honestly. I wanted to leave existing code as untouched as possible.